embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt

```diff
@@ -1,9 +1,60 @@
 package io.embrace.android.embracesdk.capture.envelope.resource
 
+import io.embrace.android.embracesdk.Embrace
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
 import io.embrace.android.embracesdk.internal.payload.EnvelopeResource
+import io.embrace.android.embracesdk.payload.AppInfo
+import io.embrace.android.embracesdk.payload.DeviceInfo
 
-internal class EnvelopeResourceSourceImpl : EnvelopeResourceSource {
+internal class EnvelopeResourceSourceImpl(
+    private val deviceInfo: DeviceInfo,
+    private val appInfo: AppInfo,
+    private val metadataService: MetadataService,
```

Comment:
IMO we have a good opportunity to get rid of the `MetadataService` by copying its existing functions into here instead. `MetadataService` could then be deleted later on when we fully switch over to the new payload.

Reply:
Done in #546